### ISSUE TITLE
no-unsafe-any and `unknown`

### ISSUE THREAD

**Issue by ThomasdenH** [palantir/tslint#4164]
Another `unknown` one:

### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const json = '{ "property": "value" }';
const result: unknown = JSON.parse(json);
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:recommended"
  ],
  "jsRules": {},
  "rules": {
    "no-unsafe-any": true
  }
}
```

#### Actual behavior
The lint triggers: `Unsafe use of expression of type 'any'.`

#### Expected behavior
Since every type is assignable to `unknown`, the usage is not actually unsafe. The lint shouldn't trigger.

**Comment by JoshuaKGoldberg** [comment]
We should note that although this visually looks similar to #3621, it's not the same. Assigning the result to `unknown` is a good idea.

**Comment by NN---** [comment]
Another sample with exception:

```ts
function g(e: unknown): void {}

function f() {
	try {

	} catch (e) {
		g(e); // it is okay to call g
	}
}
``
